modules/hostap/src/supp_api.c

```diff
@@ -511,8 +511,17 @@ static int wpas_add_and_config_network(struct wpa_supplicant *wpa_s,
 	}
 
 	/* enable and select network */
-	if (!wpa_cli_cmd_v("enable_network %d", resp.network_id)) {
-		goto out;
+	if (mode_ap)
+	{
+		if (!wpa_cli_cmd_v("enable_network %d", resp.network_id)) {
+			goto out;
+		}
+	}
+	else
+	{
+		if (!wpa_cli_cmd_v("select_network %d", resp.network_id)) {
```

Comment:
for both AP and STA you can do `enable` and `select` for both.

Reply:
Hi @krish2718 
the setup_interface_monitoring() already call net_if_foreach(iface_cb, ctx), and iface_cb will add_interface.

Reply:
Hi @krish2718 
but select_network is enough for sta to connect, no need the enable_network.
I add the distinguish between sta and AP, because I don't want to change the logic of AP mode, only want to change for sta mode.

Reply:
will update it